Agents.md

```diff
@@ -0,0 +1,263 @@
+# AI Agents Guide for dd-sdk-ios
+
+> This document outlines the rules, constraints, and expectations for AI agents contributing to the Datadog iOS SDK. Because the SDK powers mission-critical telemetry for thousands of customer applications, all changes must preserve stability, compatibility, and performance. Agents must strictly follow the guidelines below when navigating the repository, modifying code, or generating pull requests.
+
+## Project Overview
+
+The SDK is responsible for reliably collecting, batching, and transmitting telemetry from customer apps under strict performance, safety, and compliance constraints.
+
+This is the **Datadog SDK for iOS and tvOS** - Swift and Objective-C libraries to interact with Datadog. The SDK enables:
+- **Logs**: Send logs to Datadog
+- **Traces**: Distributed tracing
+- **RUM**: Real User Monitoring events collection
+- **Session Replay**: Visual session recording
+- **Crash Reporting**: Crash detection and reporting
+- **WebView Tracking**: Hybrid app monitoring
+- **Feature Flags**: Feature flag integration
+
+## Related Documentation
+
+Before starting work, familiarize yourself with these key documents:
+
+- **`docs/LLM_FEATURE_DOCS_GUIDELINES.md`** - Guidelines for creating and updating `*_FEATURE.md` files
+- **`ZEN.md`** - Core SDK philosophy and principles
+- **`CONTRIBUTING.md`** - General contribution guidelines
+- **Feature-specific docs** - Each module has a `*_FEATURE.md` file (e.g., `DatadogRUM/RUM_FEATURE.md`)
+
+## Critical: Module Architecture
+
+The SDK is organized as a **modular monorepo**. Understanding module dependencies is crucial:
+
+```
+DatadogInternal (shared protocols, types, utilities)
+       ↑
+       ├── DatadogCore (SDK initialization, data pipeline, networking)
+       │        ↑
+       │        ├── DatadogLogs
+       │        ├── DatadogTrace
+       │        ├── DatadogRUM
+       │        ├── DatadogSessionReplay
+       │        ├── DatadogCrashReporting
+       │        ├── DatadogWebViewTracking
+       │        └── DatadogFlags
+```
+
+- Feature modules MUST NOT import each other
+- Only DatadogCore may orchestrate feature lifecycles
+- DatadogInternal is the ONLY allowed place for shared types
+
+### IMPORTANT: Call Site Synchronization
+
+**When modifying code in feature modules (Logs, Trace, RUM, etc.), you MUST check and update corresponding call sites in `DatadogCore` and `DatadogInternal`.**
+
+Common oversight:
+Agents modify a module's interface or behavior but forget to:
+- Update how `DatadogCore` initializes or registers the feature
```

Comment:
`DatadogCore` does not "initialize" features so this wording sounds confusing 🤔 💭. Features are initialized directly from `<Feature>.enable()` call and get registered to the core.